src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php

```diff
@@ -302,31 +300,29 @@ protected function describeEventDispatcherListeners(EventDispatcherInterface $ev
         $this->writeText($this->formatSection('event_dispatcher', $label)."\n", $options);
 
         $registeredListeners = $eventDispatcher->getListeners($event);
+        $table = new Table($this->output);
+
         if (null !== $event) {
             $this->writeText("\n");
-            $table = new TableHelper();
             $table->setHeaders(array('Order', 'Callable'));
 
             foreach ($registeredListeners as $order => $listener) {
                 $table->addRow(array(sprintf('#%d', $order + 1), $this->formatCallable($listener)));
             }
-
-            $this->renderTable($table);
         } else {
             ksort($registeredListeners);
             foreach ($registeredListeners as $eventListened => $eventListeners) {
                 $this->writeText(sprintf("\n<info>[Event]</info> %s\n", $eventListened), $options);
 
-                $table = new TableHelper();
                 $table->setHeaders(array('Order', 'Callable'));
 
                 foreach ($eventListeners as $order => $eventListener) {
                     $table->addRow(array(sprintf('#%d', $order + 1), $this->formatCallable($eventListener)));
                 }
-
-                $this->renderTable($table);
             }
         }
+
+        $table->render();
```

Comment:
Moving this here means you will output a table with headers but without any content rows. It's not exactly the same behavior.